podman_crun/README.md

```diff
@@ -0,0 +1,124 @@
+
+```bash
+wget -qO- https://raw.githubusercontent.com/second-state/wasmedge-containers-examples/main/podman_crun/simple_wasi_application.sh | bash
+```
+
+You should see results from the WebAssembly program printed in the console log. [Here is an example](https://github.com/second-state/wasmedge-containers-examples/runs/4321868699?check_suite_focus=true#step:4:63).
+
+The sections below are step-by-step instructions for the above demo.
+## Install podman
+```
+sudo apt-get -y update
+sudo apt-get -y install podman 
+```
+
+## Install WasmEdge
+
+Use the [simple install script](https://github.com/WasmEdge/WasmEdge/blob/master/docs/install.md) to install WasmEdge.
+
+```bash
+wget -qO- https://raw.githubusercontent.com/WasmEdge/WasmEdge/master/utils/install.sh | bash -s -- -p /usr/local
+```
+
+## Build and install crun
+
+You need a `crun` binary that supports WasmEdge. For now, the easiest approach is just built it yourself from source. First, let's make sure that `crun` dependencies are installed on your Ubuntu 20.04.
+For other Linux distributions, please [see here](https://github.com/containers/crun#readme).
+
+```bash
+sudo apt update
+sudo apt install -y make git gcc build-essential pkgconf libtool \
+   libsystemd-dev libprotobuf-c-dev libcap-dev libseccomp-dev libyajl-dev \
+   go-md2man libtool autoconf python3 automake
+```
+
+Next, configure, build, and install a `crun` binary with WasmEdge support.
+
+```bash
+git clone https://github.com/containers/crun
+cd crun
+./autogen.sh
+./configure --with-wasmedge
+make
+sudo make install
+```
+
+## Install containerd
+
+Use the following commands to install containerd on your system.
+
+```bash
+export VERSION="1.5.7"
+echo -e "Version: $VERSION"
+echo -e "Installing libseccomp2 ..."
+sudo apt install -y libseccomp2
+echo -e "Installing wget"
+sudo apt install -y wget
+
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+sha256sum --check cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+
+sudo tar --no-overwrite-dir -C / -xzf cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+sudo systemctl daemon-reload
+sudo systemctl start containerd
+```
+
+## Configure containerd
+
+```
+# TODO: add containerd configuration for runc (for demo no extra config needed)
+```
+
+## Run a simple WebAssembly app
+
+Finally, we can run a simple WebAssembly program using containerd.
+[A seperate article](../simple_wasi_app.md) explains how to compile, package, and publish the WebAssembly
+program as a container image to Docker hub.
+In this section, we will start off pulling this WebAssembly-based container
+image from Docker hub using podman.
+
+```bash
+sudo podman pull docker.io/zjujinchen/mywasm-image:latest
```

Comment:
We should use the same image for all demos.

Reply:
Actually, this image that I builded is familiar for me. Of course, I just try the image builded by hydai. It works as well, I will use the same images and commits again.